### PR TITLE
Update v1.1.0

### PR DESCRIPTION
**Pull request by zmccaw-insitro**
Update based on a feature request from @dvg-p4. 
* `COAST` and `COASTSS` have been updated to allow for an arbitrary number of discrete annotation categories.
    - As part of this change, the expected annotation category labels have changed from `c(0, 1, 2)` to `c(1, 2, ..., L)`, where  `L` is the total number of annotation categories. The 1-indexed labels are more consistent with how indexing is done in R. The main functions still support 0-indexed labels. 
* The data generating process `DGP` has likewise been updated to allow for simulation from different numbers of annotation categories.
* The README and NEWS have been updated to reflect these changes. 
* Additional vignettes were added on COAST-SS and the DGP.
* Documentation and defaults have been updated for several functions.
* The updated code passes `R CMD CHECK`.

### CODE REVIEW

**Comment by zmccaw-insitro** [comment]
* For `CalcSumstats`, moved both the annotations `anno` and `maf` into the `sumstats` data frame, such that all variant-level information is together.
* Updated the default `pval_weights` of `COASTSS` to sum to 1, which is more intuitive.

File: R/allelic_series_sumstats.R [hunk 1]
```diff
@@ -250,9 +268,10 @@ ASKATSS <- function(
 #' @param maf (snps x 1) vector of minor allele frequencies. Although ideally
 #'   provided, defaults to the zero vector.
 #' @param pval_weights (3 x 1) vector of relative weights for combining the
-#'   component tests to perform the omnibus test.
-#' @param weights (3 x 1) vector of annotation category weights. The default of
+#'   component tests to perform the omnibus test. The default of
 #'   c(1, 1, 2) gives the SKAT test equal weight to the two burden tests.
```
Comment (tsoare-insitro) [inline R/allelic_series_sumstats.R:272]: c(0.25, 0.25, 0.5)

File: vignettes/coast_ss.Rmd [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+show(results)
+```
+
+### Test options
+
+* `eps` is a regularization term added to the diagonal of the LD matrix if the provided LD matrix is not positive definite. The default value is `eps = 1`. Larger values may be needed, but smaller values are not recommended.
+
+* `lambda` is an optional 3 x 1 vector of inflation factors that are applied to the p-values of the `baseline`, `sum_count`, and `allelic_skat` tests *before* the omnibus p-value is calculated. By default, `lambda = c(1, 1, 1)`, which results in no correction. Larger values may be needed, particularly if more-common variants are included. Values less than 1 will be reset to 1.
+
+* `pval_weights` is a 3 x 1 vector specifying the relative weights of the p-values from the `baseline`, `sum_count`, and `allelic_skat` tests when calculating the omnibus p-value. By default, `pval_weights = c(1, 1, 2)`, which gives the allelic SKAT test equal weight to the two burden-type tests (i.e. the baseline and allelic sum tests).
```
Comment (tsoare-insitro) [inline vignettes/coast_ss.Rmd:143]: c(0.25, 0.25, 0.5)

**Review by tsoare-insitro — APPROVED** [review]
LGTM

File: vignettes/coast_ss.Rmd (hunk 2 above)
Comment (zmccaw-insitro) [reply vignettes/coast_ss.Rmd:143]: Yes, thanks!

**Pull request merged**